fl4health/utils/losses.py

```diff
@@ -2,15 +2,103 @@
+        """
+        loss_dict: Dict[str, float] = {}
+
+        if self.additional_losses is not None:
+            for key, val in self.additional_losses.items():
+                loss_dict[key] = float(val.item())
+
+        return loss_dict
+
+    @staticmethod
+    @abstractmethod
+    def aggregate(loss_meter: LossMeter) -> Losses:
+        """
+        Aggregates the losses in the given LossMeter into an instance of Losses
+
+        Args:
+            loss_meter (LossMeter): The loss meter object with the collected losses.
+
+        Raises:
+            NotImplementedError: To be implemented by child classes.
+        """
+        raise NotImplementedError
+
+
+class EvaluationLosses(Losses):
+    def __init__(self, checkpoint: torch.Tensor, additional_losses: Optional[Dict[str, torch.Tensor]] = None) -> None:
+        """
+        A class to store the checkpoint and additional_losses of a model
+        along with a method to return a dictionary representation.
+
+        Args:
+            checkpoint (torch.Tensor): The loss used to checkpoint model (if checkpointing is enabled).
+            additional_losses (Optional[Dict[str, torch.Tensor]]): Optional dictionary of additional losses.
+        """
+        super().__init__(additional_losses)
+        self.checkpoint = checkpoint
+
+    def as_dict(self) -> Dict[str, float]:
+        """
+        Produces a dictionary representation of the object with all of the losses.
+
+        Returns:
+            Dict[str, float]: A dictionary with the checkpoint loss, plus each one of the keys in
+                additional losses if they exist.
+        """
+        loss_dict = super().as_dict()
+        loss_dict["checkpoint"] = float(self.checkpoint.item())
+        return loss_dict
+
+    @staticmethod
+    def aggregate(loss_meter: LossMeter) -> EvaluationLosses:
+        """
+        Aggregates the losses in the given LossMeter into an instance of EvaluationLosses
+
+        Args:
+            loss_meter (LossMeter): The loss meter object with the collected losses.
+
+        Returns:
+            EvaluationLosses: An instance of EvaluationLosses with the aggregated losses.
+        """
+        assert all([isinstance(losses, EvaluationLosses) for losses in loss_meter.losses_list])
+        checkpoint_loss = torch.sum(
+            torch.FloatTensor([losses.checkpoint for losses in loss_meter.losses_list])  # type: ignore
+        )
+        if loss_meter.get_type() == LossMeterType.AVERAGE:
+            checkpoint_loss /= len(loss_meter.losses_list)
+
+        additional_losses_list = [losses.additional_losses for losses in loss_meter.losses_list]
+        additional_losses_dict = _aggregate_losses_dict(additional_losses_list, loss_meter.get_type())
+
+        return EvaluationLosses(checkpoint=checkpoint_loss, additional_losses=additional_losses_dict)
+
+
+class TrainingLosses(Losses):
     def __init__(
```

Comment:
Very minor documentation detail flagging just in case: "A class to store the checkpoint, backward and additional_losses of a model along with a method to return a dictionary representation." 

We can remove checkpoint because the new TrainingLosses has removed that from what I can see